src/js/player.js

```diff
@@ -2435,10 +2557,12 @@ class Player extends Component {
     }
 
     // intial sources
+    this.cache_.previousSources = this.cache_.sources;
```

Comment:
We have to store `previousSource` and `previousSources` due to source setting being async in here. Without it, it was possible for the type of the source to be lost. I think we may eventually want to remove the source cache from here and only update it on `sourceset`.

```
player.src('src1.mp4');
player.src('src2.mp4');
```